dialogue-core/src/main/java/com/palantir/dialogue/core/DeadlineAdvertisementChannel.java

```diff
@@ -30,22 +31,32 @@ final class DeadlineAdvertisementChannel implements Channel {
 
     private final Channel delegate;
     private final Duration readTimeout;
+    private final Deadlines.Enforcement enforcement;
 
-    DeadlineAdvertisementChannel(Channel delegate, Duration readTimeout) {
+    static DeadlineAdvertisementChannel create(Channel delegate, Duration readTimeout) {
+        return new DeadlineAdvertisementChannel(delegate, readTimeout, Deadlines.Enforcement.DEFER);
+    }
+
+    static DeadlineAdvertisementChannel create(Channel delegate, Duration readTimeout, Enforcement enforcement) {
+        return new DeadlineAdvertisementChannel(delegate, readTimeout, enforcement);
```

Comment:
see my [other comment](https://github.com/palantir/dialogue/pull/2681/files#r2414110093) re: modeling as optional.
```suggestion
    static DeadlineAdvertisementChannel create(Channel delegate, Duration readTimeout, Optional<Boolean> maybeEnforced) {
        Deadlines.Enforcement enforcement = maybeEnforced.map(v -> v ? Deadlines.Enforcement.ENFORCE : Deadlines.Enforcement.DISABLE).orElse(Deadlines.Enforcement.DEFER);
        return new DeadlineAdvertisementChannel(delegate, readTimeout, enforcement);
```